Python/Module6_Testing/Hypothesis_Practice_Exercises.md

```diff
@@ -0,0 +1,232 @@
+---
+jupyter:
+  jupytext:
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.2'
+      jupytext_version: 1.3.0
+  kernelspec:
+    display_name: Python [conda env:scicomp]
+    language: python
+    name: conda-env-scicomp-py
+---
+
+<!-- #raw raw_mimetype="text/restructuredtext" -->
+.. meta::
+   :description: Topic: Practice exercises using Hypothesis, Difficulty: Easy, Category: Section
+   :keywords: hypothesis, practice, exercise, test  
+<!-- #endraw -->
+
+# Practice Exercises Using Hypothesis
+
+Hypothesis will not only improve the quality of our tests, but it should also save us time and cognitive load as it simplifies the process for describing the data that we want to pass to our code.
+That being said, it can take some practice to learn one's way around [Hypothesis' core strategies](https://hypothesis.readthedocs.io/en/latest/data.html#core-strategies);
+thus this section is dedicated to providing some useful exercises towards this end.
+
+
+<div class="alert alert-info">
+
+**Exercise: Describing data with `st.lists`**
+
+Write a strategy that generates lists of even-valued integers, ranging from length-0 to length-10. 
+
+Write a test that checks these properties and run the test.
+
+</div>
+
+
+
+<div class="alert alert-info">
+
+**Exercise: Using Hypothesis to learn about floats.. Part 1**
+
+Use the `st.floats` strategy to identify which float(s) violate the identity: `x == x`.
```

Comment:
```suggestion
Use the `st.floats()` strategy to identify which float(s) violate the identity: `x == x`.
```

(and so on throughout the PR)

This might be on the wrong side of pedantry, but I've always thought that using terminology exquisitely correctly in front of students is pretty important even if you don't explicitly define it. In this case, `st.floats is more precisely "a function which returns a strategy", while `st.floats()` *is* a strategy. The simple option is to just always include the parens, as Sphinx would.

Reply:
> I've always thought that using terminology exquisitely correctly in front of students is pretty important even if you don't explicitly define it.

I totally agree, and I think that this is an important distinction! Thanks for this.